tests/unit/test_utils_filesystem.py

```diff
@@ -0,0 +1,62 @@
+import os
+import shutil
+
+import pytest
+
+from pip._internal.utils.filesystem import copy2_fixed, is_socket
+
+from ..lib.filesystem import make_socket_file, make_unreadable_file
```

Comment:
```suggestion
from tests.lib.filesystem import make_socket_file, make_unreadable_file
```

nit: The absolute import is clearer IMO.

Reply:
I prefer relative in general, but it looks like the majority of the test code is using absolute so I will change this. :+1:

Reply:
Updated [here](https://github.com/pypa/pip/pull/6844/files#diff-e5e3005ef6c051e6287f93d9c1c721e3R8).